exchanges/exchange.go

```diff
@@ -340,6 +340,11 @@ func (b *Base) GetName() string {
 	return b.Name
 }
 
+// SetName is a method that overrides the name of the exchange base
+func (b *Base) SetName(name string) {
+	b.Name = name
```

Comment:
Given GRPC's asynchronous nature and how much we reference exchange's by their names, having the ability to change an exchange's name at any time, this function introduces data races. 

There are a couple of options:
- One can already modify exchange names via `b := e.GetBase()`, `b.Name = "wow!"`. It doesn't solve the data race, but `SetName` isn't really an option I want to export to the interface without protection.
- You may instead wish to modify setup of exchanges to allow the name to be changed at that stage and never again. 
  - eg in your interface `CustomExchangeBuilder`, modify `NewExchangeByName(name string)` to have `NewExchangeByName(referenceName, modifiedExchangeName string)`.  
- Otherwise, introducing locks around getting setting the exchange name sounds far more arduous but offers race protection

Reply:
Seeing that this is a very specific use case maybe it's best to keep things simple and instead only offer the `b := e.GetBase(), b.Name = "wow!"`option for now and revisit if similar scenario comes along.

Reply:
@lrascao I agree with that option for now, as well.